desc/integrals/surface_integral.py

```diff
@@ -0,0 +1,723 @@
+"""Surface integrals of non-singular functions.
+
+The API is subject to change.
+"""
+
+from desc.backend import cond, fori_loop, jnp, put
+from desc.grid import ConcentricGrid, LinearGrid
+from desc.utils import errorif, warnif
+
+# TODO: Make the surface integral stuff objects with a callable method instead of
+#       returning functions. Would make simpler, allow users to actually see the
+#       docstrings of the methods, and less bookkeeping to default to more
+#       efficient methods on tensor product grids.
```

Comment:
I know JAX style is functional programming, but i think OOP is better

Reply:
yeah I've been seeing similar things on #1043. Probably less of an issue here since the local functions are usually created under jit themselves so they won't cause a lot of recompilation but classes with jitted methods also work well.